### PR TITLE
chore(ts): fix typescript type-checking errors

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Related to #689 

## Description of the change:

Added Typescript type-checking to ensure correct typings. Also, run this type checking in CI on pull request.

As for Mirage JS, I kept the type as `any` to represent unknown shapes of parsed objects returned from API server.

## Motivation for the change:

Help catch mismatched types when defining interfaces, types and creating object. This helps avoiding potential bugs (i.e. a field missing or extra fields are added).

### CODE REVIEW

**Comment by tthvo** [comment]
Also, sorta odd that Mbean Metrics snapshots were removed in the source code. Added back now.

**Comment by tthvo** [comment]
References: https://github.com/patternfly/patternfly-react-seed/pull/94

File: package.json [hunk 1]
```diff
@@ -33,7 +33,8 @@
     "build:bundle-analyze": "webpack-bundle-analyzer ./stats.json dist",
     "build:bundle-profile-analyze": "concurrently -c 'auto' 'yarn:build:bundle-*'",
     "yarn:install": "yarn install",
-    "yarn:frzinstall": "yarn install --immutable"
+    "yarn:frzinstall": "yarn install --immutable",
+    "type-check": "tsc --noEmit"
```
Comment (andrewazores) [inline package.json:37]: Maybe also add this as a check that runs on `lint`?

File: src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters.tsx [hunk 2]
```diff
@@ -53,14 +52,13 @@ export interface AutomatedAnalysisFiltersProps {
...
-  const { t } = useTranslation();
 
   const currentCategory = useSelector((state: RootState) => {
-    const targetAutomatedAnalysisFilters = state.automatedAnalysisFilters.state.targetFilters.filter(
+    const targetAutomatedAnalysisFilters = state.automatedAnalysisFilters.targetFilters.filter(
       (targetFilter) => targetFilter.target === props.target
     );
     if (!targetAutomatedAnalysisFilters.length) {
-      return t('FILTER_NAME', { ns: 'common' });
+      return 'Name';
```
Comment (andrewazores) [inline src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters.tsx:61]: Why does this need to be changed this way? Is this not actually a display string but an internal lookup key?

Comment (tthvo) [reply src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters.tsx:61]: Right exactly: https://github.com/cryostatio/cryostat-web/blob/8b8de35adfb0951efc7c3de1b7f6883c2e940044/src/app/Shared/Redux/Filters/AutomatedAnalysisFilterSlice.tsx#L199

Comment (tthvo) [reply src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters.tsx:61]: Previously, using locale will cause the filters to be lost every time language is changed. Tho, we only have `en` for the moment.

Comment (tthvo) [reply src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters.tsx:61]: But seems like its also used as display. I will fix that.

Comment (tthvo) [reply src/app/Dashboard/AutomatedAnalysis/AutomatedAnalysisFilters.tsx:69]: Fixed now :)) Thanks!

**Review by andrewazores — APPROVED** [review]

**Pull request merged**